core/src/services/routing.js

```diff
@@ -455,12 +455,18 @@ export const navigateTo = async route => {
   window.dispatchEvent(event);
 };
 
-export const buildFromRelativePath = path => {
-  if (LuigiConfig.getConfigValue('routing.useHashRouting')) {
-    return window.location.hash + '/' + path;
-  } else {
-    return window.location.pathname + '/' + path;
+export const buildFromRelativePath = node => {
+  let windowPath = LuigiConfig.getConfigValue('routing.useHashRouting')
+    ? window.location.hash
+    : window.location.pathname;
+  if (node.parent && node.parent.pathSegment) {
+    windowPath = windowPath.substr(
```

Comment:
Even tough I am a bit confused what this code is doing, it works. Maybe you want to add a comment what you are achieving here, since it is quite hard to read.